make/modules/jdk.compiler/Gendata.gmk

```diff
@@ -36,58 +40,53 @@ $(eval $(call IncludeCustomExtension, modules/jdk.compiler/Gendata.gmk))
 $(eval $(call ReadImportMetaData))
 
 # Modules that should be visible for 9 - the documented modules:
-CT_MODULES := $(DOCS_MODULES)
+CT_MODULES := $(filter-out $(MODULES_FILTER), $(DOCS_MODULES))
 
 # Get the complete module source path:
 CT_MODULESOURCEPATH := $(call GetModuleSrcPath)
 
 CT_DATA_DESCRIPTION += $(MODULE_SRC)/share/data/symbols/symbols
 
 COMPILECREATESYMBOLS_ADD_EXPORTS := \
-    --add-exports java.base/jdk.internal.javac=java.compiler.interim,jdk.compiler.interim \
-    --add-exports jdk.internal.opt/jdk.internal.opt=jdk.compiler.interim,jdk.javadoc.interim \
-    --add-exports jdk.compiler.interim/com.sun.tools.javac.api=ALL-UNNAMED \
-    --add-exports jdk.compiler.interim/com.sun.tools.javac.code=ALL-UNNAMED \
-    --add-exports jdk.compiler.interim/com.sun.tools.javac.util=ALL-UNNAMED \
-    --add-exports jdk.compiler.interim/com.sun.tools.javac.jvm=ALL-UNNAMED \
+    --add-modules jdk.compiler,jdk.jdeps \
+    --add-exports jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED \
+    --add-exports jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED \
+    --add-exports jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED \
+    --add-exports jdk.compiler/com.sun.tools.javac.jvm=ALL-UNNAMED \
+    --add-exports jdk.jdeps/com.sun.tools.classfile=ALL-UNNAMED \
     #
 
+# TODO: Unify with jdk.javadoc-gendata. Should only compile this once and share.
 $(eval $(call SetupJavaCompilation, COMPILE_CREATE_SYMBOLS, \
-    TARGET_RELEASE := $(TARGET_RELEASE_BOOTJDK), \
-    SRC := $(TOPDIR)/make/langtools/src/classes \
-        $(TOPDIR)/src/jdk.jdeps/share/classes, \
+    TARGET_RELEASE := $(TARGET_RELEASE_NEWJDK), \
+    COMPILER := buildjdk, \
+    SRC := $(TOPDIR)/make/langtools/src/classes, \
     INCLUDES := build/tools/symbolgenerator com/sun/tools/classfile, \
-    BIN := $(BUILDTOOLS_OUTPUTDIR)/create_symbols, \
+    BIN := $(BUILDTOOLS_OUTPUTDIR)/create_symbols_javac, \
     DISABLED_WARNINGS := options this-escape, \
     JAVAC_FLAGS := \
-        $(INTERIM_LANGTOOLS_ARGS) \
         $(COMPILECREATESYMBOLS_ADD_EXPORTS), \
 ))
 
 $(SUPPORT_OUTPUTDIR)/symbols/ct.sym: \
     $(COMPILE_CREATE_SYMBOLS) \
     $(wildcard $(MODULE_SRC)/share/data/symbols/*) \
-    $(MODULE_INFOS)
+    $(MODULE_INFOS) \
+    $(foreach m, $(CT_MODULES) $(call FindTransitiveIndirectDepsForModules, $(CT_MODULES)), \
```

Comment:
This dependency line is hard to read. Maybe extract it to a variable and depend on the contents of the variable instead?

Also, this is the only place where CT_MODULES is used. I'd recommend also creating an intermediate variable like CT_TRANSITIVE_MODULES for $(call FindTransitiveIndirectDepsForModules, $(CT_MODULES)), and grouping all these declarations in the same place (either in the top of the file as now, or closer to the usage here).